src/api/app/jobs/fetch_local_package_version_for_project_job.rb

```diff
@@ -0,0 +1,11 @@
+class FetchLocalPackageVersionForProjectJob < ApplicationJob
+  queue_as :default
+
+  def perform(project_name)
+    project = Project.find_by_name(project_name)
+
+    project.packages.each do |package|
+      FetchLocalPackageVersionJob.perform_later(project.name, package.name)
```

Comment:
A job spawning jobs? Thousands for a usual distribution? Not such a good idea IMHO. At least not without your own queue.